pkg/operator/informers.go

```diff
@@ -18,12 +18,69 @@ import (
 	"fmt"
 
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
 
 	"github.com/prometheus-operator/prometheus-operator/pkg/informers"
 	"github.com/prometheus-operator/prometheus-operator/pkg/k8sutil"
 )
 
+type InformerGetter interface {
+	GetInformers() []informers.InformLister
+}
+
+type RefResolver interface {
+	HasRefTo(namespacedName string, obj runtime.Object) bool
+}
+
+// HasReferenceFunc returns a function which takes a object (secret or
+// configmap) as input parameter and returns true if at least one workload
+// watched by the controller has a reference to this object or is in the same
+// namespace.
+//
+// The object is expected to be a [*metav1.PartialObjectMetadata].
+func HasReferenceFunc(
+	informerGetter InformerGetter,
+	refResolver RefResolver,
+) FilterFunc {
+	return func(obj interface{}) bool {
+		partialObjMeta, ok := obj.(*metav1.PartialObjectMetadata)
+		if !ok {
+			return false
+		}
+
+		for _, informer := range informerGetter.GetInformers() {
+			workloads, err := informer.Lister().List(labels.Everything())
```

Comment:
do we need debug log for error here or would it be noisy?

Reply:
I fear that it would be noisy indeed.